### PR TITLE
UI-3314: Tab system on port listing

### PR DESCRIPTION
**Pull request by masmerino13**

### CODE REVIEW

File: src/apps/common/submodules/portListing/portListing.js [hunk 1]
```diff
@@ -65,18 +67,45 @@ define(function(require) {
 			);
 		},
 
+		portListingRenderLayout: function() {
+			var self = this,
+				subTabs = _.map(self.appFlags.portListing.subtabs, function(tab) {
+					return {
+						text: self.i18n.active().portListing.tabs.subtabs[tab],
+						id: tab,
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/portListing/portListing.js]: Yep, my plan is generate/filter ports based on those id's

File: src/apps/common/submodules/portListing/portListing.js [hunk 2]
```diff
@@ -65,18 +67,45 @@ define(function(require) {
...
+						menus: [{
+							tabs: subTabs
+						}]
+					};
+				});
+
+			monster.ui.generateAppLayout(self, {
+				menus: [
+					{
+						tabs: monster.util.isMasquerading() ? subTabs : tabs
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/portListing/portListing.js]: that means if the user isn't masquerading but isn't a *reseller* only the first 3 subtabs (from Account tab) should be displayed, is that right? < @azefiel

File: src/apps/common/submodules/portListing/portListing.js [hunk 3]
```diff
@@ -65,18 +67,45 @@ define(function(require) {
...
 		portListingRender: function(args) {
 			var self = this,
-				isMonsterApp = _.isBoolean(args.isMonsterApp)
-					? args.isMonsterApp
-					: false,
-				container,
-				parent;
+				isMonsterApp = _.isBoolean(args.isMonsterApp) ? args.isMonsterApp : false,
+				parent,
+				container;
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/portListing/portListing.js]: haha, I think the line is short and more easy to read in a single line

File: src/apps/common/submodules/portListing/portListing.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (azefiel) [reply src/apps/common/submodules/portListing/portListing.js]: That's right.

[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/portListing/portListing.js]: got it, thanks!

File: src/apps/common/i18n/en-US.json [hunk 4]
```diff
@@ -755,6 +755,16 @@
 			"textarea": {
 				"placeholder": "New Message..."
 			}
+		},
+		"tabs": {
+			"account": "Account",
+			"agent": "Agent",
+			"descendants": "Descendants"
+		},
+		"subtabs": {
```
Comment (azefiel) [inline src/apps/common/i18n/en-US.json]: I meant that you should store all tab related strings under the same object, so you can move everything from `subtabs` under `tabs`.

File: src/apps/common/submodules/portListing/portListing.js [hunk 5]
```diff
@@ -75,8 +77,24 @@ define(function(require) {
 				isMonsterApp = _.isBoolean(args.isMonsterApp)
 					? args.isMonsterApp
 					: false,
+				parent,
```
Comment (azefiel) [inline src/apps/common/submodules/portListing/portListing.js]: Please declare unassigned variables last.

Generally, this is helpful when later on you might need to assign a variable depending on one of the previous assigned variables.

**Pull request merged**